Makefile

```diff
@@ -24,25 +31,61 @@ NVCCFLAGS= -std=c++14 -ccbin=$(CXX) -O3 $(NVARCH) -Wno-deprecated-gpu-targets \
 # and enable cufinufft internal flags.
 #NVCCFLAGS+= -DINFO -DDEBUG -DRESULT -DTIME
 
-# CUDA Related build dependencies
-CUDA_ROOT=/usr/local/cuda
-INC=-I$(CUDA_ROOT)/include \
-	-Icontrib/cuda_samples
-NVCC_LIBS_PATH=-L$(CUDA_ROOT)/lib64
+# CUDA Related build dependencies -- the user can overwrite CUDA_ROOT using the
+# CUDA_DIR environment variable. If neither (CUDA_ROOT, nor CUDA_DIR) is set,
+# CUDA_ROOT defaults to /usr/local/cuda
+ifeq ($(CUDA_DIR),)
+    CUDA_ROOT ?= /usr/local/cuda
+else
+    CUDA_ROOT := $(CUDA_DIR)
+endif
+
+# Common includes
+INC=-I$(CUDA_ROOT)/include -Icontrib/cuda_samples
+ifdef FFTW_INC
+    $(info detected a FFTW_INC variable -- setting FFTW include directory)
+    INC += -I$(FFTW_INC)
+endif
 
 FFTWNAME=fftw3
 
+# Common libs
 LIBS=-lm -lcudart -lstdc++ -lnvToolsExt -lcufft -lcuda -l$(FFTWNAME) -l$(FFTWNAME)f
+ifdef FFTW_DIR
+    $(info detected a FFTW_DIR variable -- setting FFTW library directory)
+    LIBS += -L$(FFTW_DIR)
+endif
+
+# NVCC-specific includes
+NVCC_INC=
+ifdef FFTW_INC
+    NVCC_INC += -I$(FFTW_INC)
```

Comment:
I think we may not be using this variable?

Reply:
It's set by the `cray-fftw` module on cori.

I can be persuaded to move that into the site-specific `.inc`s

Reply:
Right, but I think maybe you intended to use `NVCC_INC` somewhere later in the makefile?

Reply:
Ah yes -- that got eaten by a merge -- and I don't think we need `NVCC_INC` anymore! I'll Delete it.

Reply:
Done @garrettwrong -- `NVCC_INC` is gone